packages/safe-promise-all/index.js

```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+const safePromiseAll = (arrP) =>
+    Promise.allSettled(arrP).then((results) => {
+        const errors = results
+            .filter(({ status }) => status === 'rejected')
+            .map(({ reason }) => reason)
+            .flat();
+
+        if (errors.length > 0) {
+            return [errors];
+        }
+
+        const values = results
+            .filter(({ status }) => status === 'fulfilled')
+            .map(({ value }) => value)
+            .flat();
+
+        return [null, values];
```

Comment:
I think it should always return `errors` which will always resolve to an array. I think an empty array representing no errors is better than `null`?